src/Stryker.Core/Stryker.Core.UnitTest/Baseline/Providers/AzureFileShareBaselineProviderTests.cs

```diff
@@ -170,6 +170,168 @@ public async Task Save_Doesnt_Call_CreateDictionary_And_AllocateFileLocation_Whe
+                "SendAsync",
+                ItExpr.Is<HttpRequestMessage>(requestMessage => requestMessage.RequestUri == expectedUploadContentUri && requestMessage.Method == HttpMethod.Put),
+                ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(new HttpResponseMessage()
+                {
+                    StatusCode = System.Net.HttpStatusCode.Created,
+                    Content = new StringContent("File created")
+                })
+                .Verifiable();
+
+            var target = new AzureFileShareBaselineProvider(options, new HttpClient(handlerMock.Object));
+
+            await target.Save(jsonReport, projectVersion);
+
+            // assert
+            handlerMock
+               .Protected()
+               .Verify(
+                "SendAsync",
+                Times.Exactly(1),
+                ItExpr.Is<HttpRequestMessage>(req =>
+                   req.Method == HttpMethod.Get
+                   && req.RequestUri == expectedGetUri),
+                ItExpr.IsAny<CancellationToken>());
+
+            handlerMock
+               .Protected()
+               .Verify(
+                "SendAsync",
+                Times.Exactly(1),
+                ItExpr.Is<HttpRequestMessage>(req =>
+                   req.Method == HttpMethod.Put
+                   && req.RequestUri == expectedCreateProjectOutputDirectoryUri),
+                ItExpr.IsAny<CancellationToken>());
+            handlerMock
+               .Protected()
+               .Verify(
+                "SendAsync",
+                Times.Exactly(1),
+                ItExpr.Is<HttpRequestMessage>(req =>
+                   req.Method == HttpMethod.Put
+                   && req.RequestUri == expectedCreateBaselinesDirectoryUri),
+                ItExpr.IsAny<CancellationToken>());
+            handlerMock
+               .Protected()
+               .Verify(
+                "SendAsync",
+                Times.Exactly(1),
+                ItExpr.Is<HttpRequestMessage>(req =>
+                   req.Method == HttpMethod.Put
+                   && req.RequestUri == expectedCreateVersionDirectoryUri),
+                ItExpr.IsAny<CancellationToken>());
+
+            handlerMock
+              .Protected()
+              .Verify(
+               "SendAsync",
+               Times.Exactly(1),
+               ItExpr.Is<HttpRequestMessage>(req =>
+                  req.Method == HttpMethod.Put
+                  && req.RequestUri == expectedFileAllocationUri
+                  && req.Headers.Contains("x-ms-type")),
+                ItExpr.IsAny<CancellationToken>());
+
+            handlerMock
+              .Protected()
+              .Verify(
+               "SendAsync",
+               Times.Exactly(1),
+               ItExpr.Is<HttpRequestMessage>(req =>
+                  req.Method == HttpMethod.Put
+                  && req.RequestUri == expectedUploadContentUri),
+                ItExpr.IsAny<CancellationToken>());
```

Comment:
For me just calling `handlerMock.VerifyAll()` is ok. This test is now a lot of lines of code. It makes it harder to maintain. The only downside of VerifyAll is that it doesn't check for Times.Once but I accept that.